### DISCUSSION TITLE
Deploying cloudlog with configuration management (e.g. ansible)

### DISCUSSION THREAD

**Discussion by Cadair** [magicbug/Cloudlog#1076] [Q&A]
Hello,

I am relatively new to the hobby and am also a bit of a self-hosting nerd so cloudlog looks right up my street. So I sat down and started writing an ansible role to automate the deployment of cloudlog onto my home server. I have the basics working, it clones the repo, makes a database and configures nginx etc. Having got this far I have some questions and an issue or two.

Ideally, what I would like to be able to do is to never use `/install` and have the role configure everything for me. So my first question is: What config files would I need to write to do this? and is this at all a supported way of operating?

Secondly, I just had to nuke my install and start again because of some fun network issues, and when I reconfigured it the database migrations raised an error because a table already existed. So is there anything I need to specifically consider or configure when making a fresh install of cloudlog ontop of an existing database?

Also any general thoughts on how to work with cloudlog in a way which is as friendly to these idempotent configuration management systems would be great.

Thanks for all the work on cloudlog!

**Comment by Cadair** [comment]
I investigated my database error a little more, the steps to reproduce are:

1. Install cloudlog and submit the /install form with details for a fresh database.
2. Delete the cloudlog directory
3. re-clone the repo
4. re-run the install wizard pointing at the database you already initialized.
5. Observe the following error:

```
A Database Error Occurred

Error Number: 1050

Table 'eQSL_images' already exists

CREATE TABLE `eQSL_images` ( `id` INT(5) UNSIGNED NOT NULL AUTO_INCREMENT, `qso_id` VARCHAR(250) NOT NULL, `image_file` VARCHAR(250) NOT NULL, `modified` timestamp NULL, CONSTRAINT `pk_eQSL_images` PRIMARY KEY(`id`) ) DEFAULT CHARACTER SET = utf8mb4 COLLATE = utf8mb4_general_ci

Filename: migrations/036_create_eqsl_images_table.php

Line Number: 30
```

**Comment by magicbug** [comment]
Just thinking about things off hand so might be not fully correct but

- /install/assets/install.sql includes the base SQL for Cloudlog including the default logins
- /install/config/ contains two files that need setup inside /application/config/ should be able to work out which fields need populating from the installer.

apart from that I don't see any issues you might face the installer doesn't do much.

**Reply by Cadair** [reply]
I found the install.sql file last night, I should be able to use that to provision a new database. The issues I have faced is that the installer and the install.sql file don't check to see if there is already a database where you point it to. which means if you run the installer twice to recreate the configuration or something your database gets put into a broken state.

Is there anything I should keep an eye on with regards to config files changing over time?

Thanks for the advice.
